week4-ts/src/pages/MainPage.tsx

```diff
@@ -0,0 +1,92 @@
+import styled from "styled-components";
+import SearchSection from "../components/SearchSection";
+import ResultSection from "../components/ResultSection";
+import { useState } from "react";
+import { Store } from "../types";
+import { getLocationBasedSearch, getStoreBasedTown } from "../services";
+
+export default function MainPage() {
+  const [storeInfo, setStoreInfo] = useState<Store[]>([]);
+  const [isChecked, setIsChecked] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onClickSearchButton = (isCheck: boolean, myTown: string) => {
+    setIsChecked(isCheck);
+    setIsLoading(true);
+
+    if (isCheck) {
+      searchLocationBased().then((res) => {
+        setStoreInfo(res);
+        setIsLoading(false);
+      });
+    } else {
+      searchTownBased(myTown).then((res) => {
+        setStoreInfo(res);
+        setIsLoading(false);
+      });
+    }
+  };
+
+  const searchLocationBased = async () => {
+    const data = await getLocationBasedSearch();
+    return data;
+  };
+
+  const searchTownBased = async (myTown: string) => {
+    const data = await getStoreBasedTown(myTown);
+    return data;
+  };
+
+  return (
+    <StWrapper>
+      <StHeader>
+        <h1>노인코래방</h1>
+      </StHeader>
+
+      <SearchSection
+        onClick={(isCheck: boolean, myTown: string) =>
+          onClickSearchButton(isCheck, myTown)
+        }
+      />
+      <ResultSection
+        storeInfo={storeInfo}
+        isCheck={isChecked}
+        isLoading={isLoading}
+      />
+    </StWrapper>
+  );
+}
+const StWrapper = styled.div`
```

Comment:
시멘틱 태그도 함 고려해보심이 어떨까요,!

```suggestion
const StWrapper = styled.main`
```

Reply:
앗! 감사합니다 눈썰미좋네요